src/services/errorService.js

```diff
@@ -14,96 +14,57 @@ See the License for the specific language governing permissions and
 
 import * as Sentry from '@sentry/react';
-import networkService from 'services/networkService';
-import { getToken } from 'actions/tokenAction';
-import { logAmount } from 'util/amountConvert';
-import JSON5 from 'json5';
 import config from 'util/config';
 
 if (config.sentry) {
   Sentry.init({ dsn: config.sentry });
 }
 
-// skip error logging
+const errorCache = [];
 const noLogErrors = [
   'user denied',
   'user rejected',
-  'server is not ready to handle the request',
-  'insufficient funds',
-  'network error',
-  'transaction has been reverted'
+  'user canceled'
 ];
 
-// used to prevent error spam from a single session
-const errorCache = [];
-
-class ErrorService {
-  log (error) {
-    // skip logging if in defined list
-    for (const i of noLogErrors) {
-      if (error.message.toLowerCase().includes(i)) {
-        return;
-      }
-    }
-
-    if (config.sentry) {
-      if (errorCache.includes(error.message)) {
-        // if message already seen in this session, ignore it
-        return;
-      }
-      errorCache.push(error.message);
-      Sentry.captureException(error);
-    }
+export class WebWalletError extends Error {
+  constructor ({
+    originalError,
+    customErrorMessage,
+    reportToSentry,
+    reportToUi
+  }) {
+    super(originalError.message);
+    this._originalError = originalError;
+    this._customErrorMessage = customErrorMessage;
+    this._reportToSentry = reportToSentry;
+    this._reportToUi = reportToUi;
   }
 
-  async sanitizeError (error) {
-    // user sign rejection from metamask
+  report (dispatchMethod) {
     if (
-      (error.message && error.message.toLowerCase().includes('user denied')) ||
-      (error.message && error.message.toLowerCase().includes('user rejected'))
+      noLogErrors.find(i => this._originalError.message && this._originalError.message.toLowerCase().includes(i)) ||
+      this._originalError.code === -32000
     ) {
-      return null;
-    }
-
-    // ignore metamask rpc header not found errors
-    if (error.code === -32000) {
-      return null;
+      return;
     }
 
-    // omg-js create UTXOs insufficient funds error
-    if (error.message && error.message.includes('Insufficient funds. Needs ')) {
-      try {
-        const tokenAddress = error.message.split(' ').find(i => i.startsWith('0x'));
-        const { decimals, name } = await getToken(tokenAddress);
-        const tokenAmount = error.message.split(' ').find((i) => {
-          return i.match(/(?!0x)\d*/)[0] !== '';
-        });
-        const decimalAmount = logAmount(tokenAmount, decimals);
-        return 'Insufficient funds. Needs '.concat(decimalAmount.toString(), ' more of ', name, ' to cover payments and fees');
-      }  catch (error) {
-        return 'Insufficient funds to cover payments and fees';
+    if (this._reportToSentry && config.sentry) {
+      if (!errorCache.includes(this._originalError.message)) {
+        errorCache.push(this._originalError.message);
+        try {
```

Comment:
does this need a try/catch?

Reply:
yeah, in the rare case the sentry report fails (ie sentry server is down, etc) so app wont crash in that case